### ISSUE TITLE
Average precision AP is decreasing !

### ISSUE THREAD

**Issue by scstu** [fizyr/keras-retinanet#1003]
it may not be a code issue, But I really need help concerning this issue? 
I have trained my model for 100 epoch. Errors are getting smaller with epochs, but the weird thing that mAP is getting smaller as well..

![training_99](https://user-images.githubusercontent.com/38878761/57719156-b0dbe200-76b1-11e9-8191-b9c29a663004.JPG)

any advice will be appriciated

**Comment by scstu** [comment]
any help please!

**Comment by LSBigum** [comment]
Sorry I can't be of much help with your issue. However, I would like to ask you how you visualize your mAP? Apologies for going off-topic.

**Comment by martinzlocha** [comment]
Well whats happening is that  your training losses are decreasing, however decreasing mAP indicates that you have started to overfit. RetinaNet doesn't generally achieve a loss lower than 0.2 so thats also an indicator that its overfitting.

To answer @LSBigum, he is using tensorboard to display the graphs.

**Comment by AllenLau9679** [comment]
> Well whats happening is that your training losses are decreasing, however decreasing mAP indicates that you have started to overfit. RetinaNet doesn't generally achieve a loss lower than 0.2 so thats also an indicator that its overfitting.
> 
> To answer @LSBigum, he is using tensorboard to display the graphs.

excuse me, when I am training, the regression loss is usually around 0.4~0.6, is this situation normal？

**Comment by hgaiser** [comment]
> it may not be a code issue, But I really need help concerning this issue?
> I have trained my model for 100 epoch. Errors are getting smaller with epochs, but the weird thing that mAP is getting smaller as well..
> 
> ![training_99](https://user-images.githubusercontent.com/38878761/57719156-b0dbe200-76b1-11e9-8191-b9c29a663004.JPG)
> 
> any advice will be appriciated

Strange, I have never really seen retinanet overfit that bad, but it does look like overfitting.

> excuse me, when I am training, the regression loss is usually around 0.4~0.6, is this situation normal？

Yes those values should be fine. It depends on the complexity of your dataset, but I would guestimate that anything in the range 0.1 - 1.0 **could** be fine.

Considering there is no real issue to discuss here, I am closing the issue.

**Comment by zhangyufei1995** [comment]
@scstu  I would like to ask you how you visualize your loss and mAP? How to write the specific code

**Comment by Amanbisht774** [comment]
@LSBigum 
%load_ext tensorboard
%tensorboard --logdir output
This will help you to create graphs
